File: backend/nodejs/src/models/Seat.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../database/connection');

const Seat = sequelize.define('Seat', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  schedule_id: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: 'schedules',
      key: 'id'
    }
  },
  seat_number: {
    type: DataTypes.INTEGER,
    allowNull: false,
    validate: {
      min: 1,
      max: 12
    }
  },
  status: {
    type: DataTypes.ENUM('available', 'booked', 'reserved'),
    defaultValue: 'available',
    allowNull: false
  },
  booking_id: {
    type: DataTypes.UUID,
    allowNull: true,
    references: {
      model: 'bookings',
      key: 'id'
    }
  }
}, {
  tableName: 'seats',
  timestamps: true,
  underscored: true,
  indexes: [
    {
      unique: true,
      fields: ['schedule_id', 'seat_number']
    }
  ]
});

module.exports = Seat;

File: backend/golang/internal/model/seat.go
package model

// Seat represents a seat in a schedule
type Seat struct {
	ID         int    `json:"id"`
	ScheduleID int    `json:"schedule_id"`
	SeatNumber int    `json:"seat_number"`
	Status     string `json:"status"` // "available", "booked", "reserved"
	BookingID  *int   `json:"booking_id,omitempty"`
}

// SeatWithBooking seat พร้อมข้อมูล booking (สำหรับ seat selection page)
type SeatWithBooking struct {
	ID            int    `json:"id"`
	ScheduleID    int    `json:"schedule_id"`
	SeatNumber    int    `json:"seat_number"`
	Status        string `json:"status"`
	BookingID     *int64 `json:"booking_id,omitempty"`
	PassengerName string `json:"passenger_name,omitempty"` // จาก booking JOIN
}
